File: Spark/simple-spark/src/main/java/org/gb/sample/spark/income/TopIncomesMain.java
package org.gb.sample.spark.income;

import java.util.List;
import java.util.Map;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class TopIncomesMain {

	public static void main(String[] args) {
		SparkConf conf = new SparkConf().setAppName("Person profiles with top incomes");
		JavaSparkContext sc = new JavaSparkContext(conf);
		String nameVsIncomeFile = args[0];
		String personProfileFile = args[1];
		JavaRDD<String> nameVsIncomeLines = sc.textFile(nameVsIncomeFile);
		JavaRDD<String> personProfileLines = sc.textFile(personProfileFile);
		IncomeAnalyzer incomeAnalyzer = new IncomeAnalyzer(nameVsIncomeLines, personProfileLines);
		Map<Integer, List<PersonProfile>> topIncomesWithPersonProfiles = incomeAnalyzer.getTopIncomePersonProfiles(10);
		topIncomesWithPersonProfiles.forEach(TopIncomesMain::printIncomeWithPersonProfiles);
		sc.close();
	}

	private static void printIncomeWithPersonProfiles(Integer income, List<PersonProfile> personProfiles) {
		personProfiles.forEach(profile -> {
			String fullName = String.join(" ", profile.getFirstName(), profile.getLastName());
			String incomeWithPersProfileAsString = String.join(", ", fullName, profile.getCompanyName(),
					profile.getEmailAddress());
			System.out.println(income + " <--> " + incomeWithPersProfileAsString);
		});
	}

}

File: Spark/simple-spark-scala/src/main/scala/org/gb/sample/spark/income/TopIncomesMain.scala
package org.gb.sample.spark.income

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object TopIncomesMain {
    def main(args: Array[String]): Unit = {
        val conf = new SparkConf().setAppName("Person profiles with top incomes")
        val sc = new SparkContext(conf)
        val nameVsIncomeFile = args(0)
        val personProfileFile = args(1)
        val nameVsIncomeLines = sc.textFile(nameVsIncomeFile)
        val personProfileLines = sc.textFile(personProfileFile)
        val incomeAnalyzer = new IncomeAnalyzer(nameVsIncomeLines, personProfileLines)
        val topIncomesWithPersonProfiles = incomeAnalyzer.getTopIncomePersonProfiles(10)
        topIncomesWithPersonProfiles.foreach(printIncomeWithPersonProfiles)
    }

    def printIncomeWithPersonProfiles(incomeWithPersProfiles: (Int, List[PersonProfile])): Unit = {
        incomeWithPersProfiles._2.foreach(persProfile => {
            val fullName = String.join(" ", persProfile.firstName, persProfile.lastName)
            val persProfileAsString = String.join(",", fullName, persProfile.companyName, persProfile.emailAddress)
            println(s"${incomeWithPersProfiles._1} <--> $persProfileAsString")
        })
    }
}